frame/support/procedural/src/pallet/expand/call.rs

```diff
@@ -140,6 +140,24 @@ pub fn expand_call(def: &mut Def) -> proc_macro2::TokenStream {
 
 	let capture_docs = if cfg!(feature = "no-metadata-docs") { "never" } else { "always" };
 
+	// Wrap all calls inside of storage layers
+	if let Some(syn::Item::Impl(item_impl)) = def
+		.call
+		.as_ref()
+		.map(|c| &mut def.item.content.as_mut().expect("Checked by def parser").1[c.index])
+	{
+		item_impl.items.iter_mut().for_each(|i| {
+			if let syn::ImplItem::Method(method) = i {
+				let block = &method.block;
+				method.block = syn::parse_quote! {{
```

Comment:
> This pr is solving that by wrapping each call function inside a storage layer  

You change the logic of an extrinsic to be wrapped inside a `with_storage_layer` here?  
So we now have the invariant that the logic inside an extrinsic is *always* transactional, no matter how its called?

Reply:
That's the idea, if you are thinking of making it configurable, there is [this PR](https://github.com/paritytech/substrate/pull/11597) that attempts to solve it.

Reply:
Yeah, we should make it configurable eventually